components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -429,6 +449,100 @@ public String exportApplication(String applicationId, Boolean exportSecrets) {
+        } catch (IdentityApplicationManagementException e) {
+            String msg = "Error exporting application with id: " + applicationId;
+            throw handleIdentityApplicationManagementException(e, msg);
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Application: " + applicationId + " exported successfully.");
+        }
+
+        return  generateFileFromModel(fileType, serviceProvider);
+    }
+
+    private TransferResource generateFileFromModel(String fileType, ServiceProvider serviceProvider) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Generating file from model: " + serviceProvider.getApplicationName());
+        }
+
+        StringBuilder fileNameSB = new StringBuilder(serviceProvider.getApplicationName());
+        String fileContent = "";
+
+        switch (fileType) {
+            case MEDIA_TYPE_APPLICATION_YAML:
+                Yaml yaml = new Yaml();
+                try {
+                    fileContent = yaml.dump(serviceProvider);
+                } catch (YAMLException e) {
+                    throw Utils.buildServerError("Error exporting application from YAML file.", e);
+                }
+                fileNameSB.append(YML_FILE_EXTENSION);
+                break;
+            case MEDIA_TYPE_APPLICATION_XML:
+                JAXBContext jaxbContext;
+                try {
+                    jaxbContext = JAXBContext.newInstance(ServiceProvider.class);
+                    Marshaller marshaller = jaxbContext.createMarshaller();
+                    marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+                    StringWriter stringWriter = new StringWriter();
+                    marshaller.marshal(serviceProvider, stringWriter);
+                    fileContent = stringWriter.toString();
+                } catch (JAXBException e) {
+                    throw Utils.buildServerError("Error exporting application from XML file.", e);
+                }
+                fileNameSB.append(XML_FILE_EXTENSION);
+                break;
+            case MEDIA_TYPE_APPLICATION_JSON:
+                ObjectMapper objectMapper = new ObjectMapper(new JsonFactory());
+                try {
+                    fileContent = objectMapper.writeValueAsString(serviceProvider);
+                } catch (JsonProcessingException e) {
+                    throw Utils.buildServerError("Error exporting application from JSON file.", e);
+                }
+                fileNameSB.append(JSON_FILE_EXTENSION);
+                break;
+            default:
+                throw Utils.buildServerError("Unsupported media type: " + fileType + "."
+                        + " Supported media types are " + MEDIA_TYPE_APPLICATION_XML + ", "
+                        + MEDIA_TYPE_APPLICATION_JSON + ", " + MEDIA_TYPE_APPLICATION_YAML + ".");
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Successfully generated file: " + fileNameSB);
```

Comment:
This would imply a file was generated at this point.
```suggestion
            log.debug("Successfully generated content for file: " + fileNameSB);
```